### ISSUE TITLE
Hidden labels within Captcha form field should not be read by screenreaders

### ISSUE THREAD

**Issue by stefan-at-work** [contao/contao#8386]
### Affected version(s)

5.*

### Description

The default Captcha form field in Contao is currently rendered without an associated label, which leads to accessibility issues. WAVE and other accessibility tools report this as an error because a form field without a label cannot be properly interpreted by screen reader users.

The corresponding label should be added and marked with a visually-hidden class to ensure it’s accessible to screen readers without being visible in the user interface.

**Comment by stefan-at-work** [comment]
Just because the captcha is never visible doesn’t mean the label isn’t currently being read out by screen readers.

**Comment by fritzmg** [comment]
I cannot confirm this. The security question does have a label.

**Comment by stefan-at-work** [comment]
Sorry, wrong description of the issue. The label is actually present and not visible visually, but in my opinion, it should be per default marked with visually-hidden so that it is also hidden from screen readers.

**Comment by fritzmg** [comment]
Screen readers should not read any content that has `display: none`, is that not the case?

**Comment by stefan-at-work** [comment]
WAVE reports the label because it is still visible in the DOM, even though the input field has display: none. That’s why my suggestion is to either hide both completely (including the label) or to mark the label with visually-hidden.

**Comment by fritzmg** [comment]
> WAVE reports the label because it is still visible in the DOM, even though the input field has display: none.

I think that's a bug in WAVE then.

**Comment by fritzmg** [comment]
But did you confirm this with an actual screen reader?

**Comment by zoglo** [comment]
WAVE and other tools parse the page and most likely do not take styles from ancestors into account.

When I've been checking forms with NVDA and Voiceover, hidden content was never read out loud (including the honeypot). However it did read out Altcha properly.

You should not trust a tool 100% as they can not replace visually impaired persons actually using screenreaders and hiding a non visible label just for a testing tool seems over the top to me - especially since not everybody includes the CSS for the visually-hidden class.

**Comment by fritzmg** [comment]
> WAVE and other tools parse the page and most likely do not take styles from ancestors into account.

And may be also styles introduced by JavaScript, which is the case here.

And yes, regardless of WAVE, I find it highly unlikely that any screen reader will ever read out any content within a `display: none` parent.

**Comment by zoglo** [comment]
I updated the title according to the wrong description:
https://github.com/contao/contao/issues/8386#issuecomment-2925534246

Since I have tested both on Mac (Voiceover) and Windows (NVDA) and both times the labels are not read (because being `display: none`), there is nothing to fix here.
